xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -214,7 +228,26 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
 
         return Arrays.asList(metadata);
     }
-
+    
+    private boolean removeBlockRecursively(Block block, Block blockToRemove)
```

Comment:
I really don't understand this method. Can you explain why you added it?

Reply:
This method will find the heading in the children as well as in the children of a child. When a child is matched with the heading block, It will remove that child and exit. The approach I added before was just looking into the child. So this method is a better implementation for removing the first heading.

Reply:
It's actually a very bad implementation IMO (you traverse all children)... and AFAICS it does nothing more than what the caller is already doing already with `result.getFirstBlock(new ClassBlockMatcher(HeaderBlock.class), Block.Axes.DESCENDANT_OR_SELF);`.  I really don't understand why you added it. You already know the block to remove!

Reply:
Yes, I agree, but when I try to remove the using `result.removeBlock(heading)` I got `Provided Block to remove is not a child` exception. `result.getFirstBlock(new ClassBlockMatcher(HeaderBlock.class), Block.Axes.DESCENDANT_OR_SELF);` will get the heading to remove. and to remove that heading we have to find in a child as well as children of a child. So that's why I implement this function. `result.removeBlock(heading)` will only look into its own children. Opinion, please.